cmake/global_variables.cmake

```diff
@@ -0,0 +1,50 @@
+# All Global variables for the top level CMakeLists.txt should be defined here
+
+# Define the Version Info
+file(READ "VERSION_INFO" ver)
+set(VERSION_INFO ${ver} CACHE STRING "Set the onnxruntime-genai version info.")
+message("Building onnxruntime-genai for version ${VERSION_INFO}")
+
+
+# Define the project directories
+set(GENERATORS_ROOT ${PROJECT_SOURCE_DIR}/src)
+set(MODELS_ROOT ${PROJECT_SOURCE_DIR}/src/models)
+set(ORT_HOME ${CMAKE_SOURCE_DIR}/ort CACHE PATH "Path to the onnxruntime root directory.")
+set(ORT_HEADER_DIR ${ORT_HOME}/include)
+set(ORT_LIB_DIR ${ORT_HOME}/lib)
+
+if(NOT EXISTS "${ORT_LIB_DIR}/${ONNXRUNTIME_LIB}")
+  message(FATAL_ERROR "Expected the ONNX Runtime library to be found at ${ORT_LIB_DIR}/${ONNXRUNTIME_LIB}. Actual: Not found.")
+endif()
+if(NOT EXISTS "${ORT_HEADER_DIR}/onnxruntime_c_api.h")
+  message(FATAL_ERROR "Expected the ONNX Runtime C API header to be found at \"${ORT_HEADER_DIR}/onnxruntime_c_api.h\". Actual: Not found.")
+endif()
+
+
+# Define the dependency libraries
+
+if(WIN32)
```

Comment:
Why the var "ONNXRUNTIME_LIB " is used before set?

Reply:
Nice catch. reoreded.